frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/ToggleListingCommitsAction.java

```diff
@@ -39,15 +40,17 @@ public void update(AnActionEvent e) {
   @Override
   @UIEffect
   public boolean isSelected(AnActionEvent e) {
-    return getGraphTable(e).isListingCommits();
+    var project = getProject(e);
+    return project.getService(GraphTableProvider.class).getGraphTable().isListingCommits();
   }
 
   @Override
   @UIEffect
   public void setSelected(AnActionEvent e, boolean state) {
     LOG.debug("Triggered with state = ${state}");
 
-    var graphTable = getGraphTable(e);
+    var project = getProject(e);
+    var graphTable = project.getService(GraphTableProvider.class).getGraphTable();
```

Comment:
Hmm move this logic to `getGraphTable` method in `IExpectsKeyProject`? same for other similar pieces (?)